app/renderer/actions/Inspector.js

```diff
@@ -142,23 +167,28 @@ export function applyClientMethod (params) {
                       getState().inspector.isRecording;
     try {
       dispatch({type: METHOD_CALL_REQUESTED});
-      let {source, screenshot, result, sourceError, screenshotError} = await callClientMethod(params.methodName, params.args, params.xpath);
+      let {source, screenshot, result, sourceError, screenshotError, 
+        variableName, variableIndex, strategy, selector} = await callClientMethod(params);
       if (isRecording) {
-        // for now just add a fake recorded step of 'finding'
-        // the element we are going to interact with. in the
-        // future we'll want to adjust this to use the locator
-        // strategy we recommend to the user, not just xpath
-        if (params.xpath) {
-          // also make sure we only do this optionally in case we're calling
-          // a global driver method that isn't operating on an element
-          recordAction('findElement', ['xpath', params.xpath])(dispatch);
+        // Add 'findAndAssign' line of code. Don't do it for arrays though. Arrays already have 'find' expression
+        if (strategy && selector && !variableIndex && variableIndex !== 0) {
+          findAndAssign(strategy, selector, variableName, false)(dispatch, getState);
```

Comment:
why pull `findAndAssign` out into its own action? it's only called here, right? unless you just wanted to make the code more modular/readable